src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1470,9 +1522,36 @@ public void ProcessFileChange(string projectSystemFilePath, string workspaceFile
 
                 ClearAndZeroCapacity(_documentsRemovedInBatch);
 
+                // Update project's order of documents.
+                solution = UpdateProjectDocumentsOrderForBatch(solution);
+
                 return solution;
             }
 
+            internal Solution UpdateProjectDocumentsOrderForBatch(Solution solution)
+            {
+                // The action of ordering documents was not included in the batch. Simply return the solution.
+                if (_orderedFilesInBatch == null)
+                {
+                    return solution;
+                }
+
+                var orderedFiles = _orderedFilesInBatch;
+
+                _orderedFilesInBatch = null;
+
+                return UpdateProjectDocumentsOrder(solution, orderedFiles);
+            }
+
+            private Solution UpdateProjectDocumentsOrder(Solution solution, IEnumerable<string> filePaths)
+            {
+                var projectId = _project.Id;
+                var documentIds =
+                    filePaths.Select(x => solution.GetDocumentIdsWithFilePath(x).Single(id => id.ProjectId == projectId));
```

Comment:
Is it easier to use _documentPathsToDocumentIds and avoid the extra .Single() there?

Reply:
(I'm guessing it's also going to be a bit cheaper too...)

Reply:
Sure, I'll look at it.